frontend/src/components/pages/UserManagement/index.tsx

```diff
@@ -15,68 +19,202 @@ import {
 import React from "react";
 
 import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import UserAPIClient from "../../../APIClients/UserAPIClient";
+import VolunteerAPIClient from "../../../APIClients/VolunteerAPIClient";
+import { Role, Status } from "../../../types/AuthTypes";
 import { DonorResponse } from "../../../types/DonorTypes";
+import { VolunteerResponse } from "../../../types/VolunteerTypes";
+import { UserMgmtTableRecord } from "./types";
 
 const UserManagementPage = (): JSX.Element => {
   const [donors, setDonors] = React.useState([] as DonorResponse[]);
+  const [volunteers, setVolunteers] = React.useState([] as VolunteerResponse[]);
+  const [users, setUsers] = React.useState([] as UserMgmtTableRecord[]); // both donors and volunteers
   const [search, setSearch] = React.useState("");
+  const [selectedFilter, setSelectedFilter] = React.useState("all");
+  const [dataChanged, setDataChanged] = React.useState<0 | 1>(0);
```

Comment:
I don't think we should use this kind of state handling to watch for if there's changes to volunteer approval or to user deletion.
By running a `useEffect` and making a whole new backend call to refresh the table for these changes then building it all over again with `buildTableUsers` is expensive.

We can delete all of:
```typescript
  React.useEffect(() => {
    const getDonors = async () => {
      const res = await DonorAPIClient.getAllDonors();
      setDonors(res);
    };

    const getVolunteers = async () => {
      const res = await VolunteerAPIClient.getAllVolunteers();
      setVolunteers(res);
    };

    getDonors();
    getVolunteers();
  }, [dataChanged]);
```
and instead for **`handleDeleteUser`** and **`handleApprove`** we filter in-place the the `users`state.
i.e. 
```typescript
   const handleDeleteUser = async (user: UserMgmtTableRecord) => {
    const deleteUserResponse = await UserAPIClient.deleteUserById(user.userId);
    if (deleteUserResponse) {
    users.filter((u)=>u.id===user.id)
}
  };

  const handleApprove =  async (user: UserMgmtTableRecord) => {
    const newVolunteerData = { status: Status.APPROVED };
    const updatedVolunteerResponse = await VolunteerAPIClient.updateVolunteerById(
      user.id,
      newVolunteerData,
    );
    users.find(u => u.id === updatedVolunteerResponse.userId).status=Status.APPROVED;

  };
```

Reply:
for the approach that Hanlin suggested for `handleApprove`, just one small addition to check for error so that we are not showing any wrong information on frontend.